### PR TITLE
3.11 - writable-tmpfs & overlay changes

### LINKED ISSUE #137: CE 3.11: document `--overlay create --sparse`

### LINKED ISSUE #138: CE 3.11: document `--overlay` unpriv where supported

### LINKED ISSUE #140: CE 3.11: document `--writable-tmpfs` unpriv where supported

### LINKED ISSUE #146: CE 3.11: document Sessiondir maxsize change to 64MiB

### PR DESCRIPTION
**Pull request by dtrudg**
## Description of the Pull Request (PR):

* --writable-tmpfs with kernel unpriv overlay
* session dir max size implication
* directory --overlay with kernel unpriv overlay
* sparse overlay creation

## This fixes or addresses the following GitHub issues:

- Fixes #137
- Fixes #138
- Fixes #140
- Fixes #146

### CODE REVIEW

File: persistent_overlays.rst [hunk 1]
```diff
@@ -87,23 +79,50 @@ So for example:
...
-Create an overlay image (< 3.8)
--------------------------------
+.. note::
 
-You can use tools like ``dd`` and ``mkfs.ext3`` to create and format an
-empty ext3 file system image, which holds all changes made in your
-container within a single file. Using an overlay image file makes it
-easy to transport your modifications as a single additional file
-alongside the original SIF container image.
+   Filesystem image overlays are only supported one {singularity} is installed in
```
Comment (EmmEff) [inline persistent_overlays.rst]:
```suggestion
   Filesystem image overlays are only supported once {singularity} is installed in
```

Did you mean "once" here?

File: singularity_and_docker.rst [hunk 2]
```diff
@@ -666,13 +666,26 @@ overlay is used and then discarded.
 
 Binding a directory into a container, or running a writable sandbox may
 or may not be safe, depending on the program executed. The program must
-use, and the filesystem support, some type of locking in order that the
 parallel runs do not interfere.
```
Comment (EmmEff) [inline singularity_and_docker.rst]: Did you mean to remove this?

**Review by EmmEff — CHANGES_REQUESTED** [review]
A couple of wording questions

File: singularity_and_docker.rst (hunk 2 above)
Comment (dtrudg) [reply singularity_and_docker.rst]: No.... whoops. Thanks.

File: persistent_overlays.rst [hunk 3]
```diff
@@ -44,13 +50,11 @@ To use a persistent overlay, you must first have a container.
...
-File system image overlay
-=========================
+Filesystem image overlay
+========================
 
 Since 3.8, {Singularity} provides a command ``singularity overlay
-create`` to create persistent overlay images. You can create a single
-EXT3 overlay image or adding a EXT3 writable overlay partition to an
-existing SIF image.
+create`` to create persistent overlay images.
```
Comment (EmmEff) [inline persistent_overlays.rst:57]: Wording...

File: persistent_overlays.rst [hunk 4]
```diff
@@ -162,9 +181,12 @@ transported or shared as easily as a single overlay file.
...
 
-   For security reasons, you must be root to use a bare directory as an
-   overlay. ext3 file system images can be used as overlays without root
-   privileges.
+   For security reasons, if {Singularity} is installed in setuid mode, you must
+   be root to use a bare directory as an overlay. ext3 file system images can be
+   used as overlays without root privileges.
+
+   Non-root users can use directory overlays if {Singularity} is install in
+   non-setuid mode, and the kernel (>=5.11) of the system supports this.
```
Comment (EmmEff) [inline persistent_overlays.rst]:
```suggestion
   Non-root users can use directory overlays if {Singularity} is installed in
   non-setuid mode, and the kernel (>=5.11) of the system supports this.
```

**Review by EmmEff — CHANGES_REQUESTED** [review]

**Review by EmmEff — APPROVED** [review]

**Pull request merged**